loki/transformations/hoist_variables.py

```diff
@@ -319,6 +330,33 @@ def driver_variable_declaration(self, routine, variables):
         """
         routine.variables += tuple(v.rescope(routine) for v in variables)
 
+    @staticmethod
+    def driver_variable_declaration_dim_remapping(routine, variables):
```

Comment:
[no action]: Just a general note: we keep hitting this remapping problem in many use cases - so eventually we might want a general utility that can do this. But for this use case this looks great.